### DISCUSSION TITLE
Supporting mnemonics written not in English

### DISCUSSION THREAD

**Discussion by ganqqwerty** [fabd/kanji-koohii#278] [1. Feedback & Feature Requests]
From what I understood, kanji-koohi mnemonics are English-only. It would be amazing to submit and use my mnemonics written in different languages. It's not a feature request, it's more like an invitation to brainstorm how the feature can look like

In this ticket I will try to justify the need and describe the target audience

I can name the following groups of people for which it's important to have their mnemonics in other languages: 

1. non-native English speakers. It's a very big group of users. For us, a lot of English mnemonics are not practical. 
2. illiterate Japanese speakers. Some people learn the language using purely auditory methods first, and start with the kanji when they are already quite advanced at speaking. They would like to write their mnemonics in Japanese using only kana. 
3. bi- and tri-lingual people, polyglots. A Latin-American person has perfect mental maps in both Spanish and English. Same goes for Turkish-German or Serbian-French speakers, and for any other combinations. Their most useful mnemonics would often use both of their native languages. 

Here are the issues that we're facing when trying to use English-only mneumonics:

1. Sometimes they as they utilize **rare words**. I still sometimes have problems with Heisig's mnemonic "cape and dagger man" for hiragana sign け. 
2 Sometimes they utilize the names of locally popular characters. Not a lot of people outside of the native English world know who Bill Cosby or Tom Brady. French speakers would like to use Charles Aznavour, Russian speakers would like to associate rich mustaches with Yakubovich. I remembered the kana ち by using a small children's poem about Chi-Chi-Chi monkey that sells bricks and suddenly farts. Every ex-USSR speaker know that stuff, lol.
3. Local [onomatopoeias](https://www.daytranslations.com/blog/cross-linguistic-onomatopoeias/). Ukrainian duck does not say quack-quack, but Ukrainian frog does. The duck says kria-kria, and the pig says khryu-khryu, so the onomatopoeia oink-oink sounds ridiculous and does not stick at all. 
4. Immediate associations using the native language. Cyrillic letter Ю looks absolutely like hiragana ゆ. Ask your Mongolian or Kazakh friend how they remembered it. At the same time, these would be useless for the native monolungual English speakers.

**Comment by fabd** [comment]
Hi @ganqqwerty thank you for the thoughtful feedback - and interesting examples.

Support for "filtering" the stories in different languages has been requested now and then over the years.

I agree , and I would love to be able to improve this - however so far I haven't thought of a simple elegant solution.

As of writing, people do share mnemonics in other languages - but the Study page doesn't do anything specific with them.

A common suggestion that I have seen before, is to add a dropdown with the language, when publishing the story. So now it gets complicated real quick... someone's not going to find their country in there - or their specific language. If the list is long, it might be a hassle to select a language. Then you'd probably want to have a "default language" setting so you don't need to set that explicitly everytime you share a story - but what about people like me who are fairly comfortable in two languages? Then this setting might not be that useful - or I might forget about it if the "default language" is already chosen, and sometimes have it incorrectly set when I publish the story.

Sorting stories by language sounds like something AI would be very good at doing - but also something the koohii servers are not prepared for - and I also personally have no experience or skills whatsoever configuring AI engines.

Still I'd love to hear ideas. If there are small steps I can do UX wise that improve the experience - we can try to do that.

**Comment by Marel561** [comment]
I would also love to have such a feature. I dont have a clue about coding but cant you send every single comment to the Google Translate API to identify the language maybe? I found this link, I dont know if it helps of if its the right one.
https://cloud.google.com/translate/docs/reference/rest/v3beta1/projects/detectLanguage

If google doesnt work, maybe another translator? I would also guess that the first 10-20 words of every story would be enough to accurately determine it. 

It would take some time but if you've got all the old ones you can simply demand the new writers to select a language.
My suggestions would be that instead of including every language you simply just include these above a certain threshold (like 50-100 stories). For example you can choose between English, Spanish, French, German (thats all the languages I saw so far I think) and "other". 
Every time "other" gets selected you send them to the API, identify them and if they reach the threshold they become a specified entry on the select list.

For the interface I would suggest a drop down menu/list of languages in the "shared stories" window. There you may either tick all languages you speak to show all related stories or just select one.  As the utmost of stories are in english I would suggest the latter as you would only see english storys anyways because they are the most upvoted. That way you would only swap to check the other languages you speak if you dont find a good one in english or the keyword is different. In this case it would make sense to me that you dont select a default language but instead the website remembers your most recent choice.

That would be my suggestions. But as I said already, I pretty much have no idea what Im talking about or if it is realistic at all.
